app/src/main/java/ch/epfl/sdp/drone3d/service/impl/location/AndroidLocationService.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.service.impl.location
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Criteria
+import android.location.LocationListener
+import android.location.LocationManager
+import ch.epfl.sdp.drone3d.service.api.location.LocationService
+import ch.epfl.sdp.drone3d.service.module.LocationModule.LocationProvider
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Location service using android location functionalities
+ */
+@Singleton
```

Comment:
I am not 100% sure that this Singleton is important.

Reply:
After checking, I also think I could remove it. I will change this on another PR.